standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1270,6 +1291,31 @@ public int getNumPartitionsViaSqlFilter(SqlFilterForPushdown filter) throws Meta
       params[i + 3] = filter.params.get(i);
     }
 
+    return getNumPartitionsViaSql(queryText, params);
+  }
+
+  public int getNumPartitionsViaSqlPs(Table table, List<String> partVals) throws MetaException {
+    String partialName = MetaStoreUtils.makePartNameMatcher(table, partVals, "_%");
+
+    // Get number of partitions by doing count on PART_ID.
+    String queryText = "select count(" + PARTITIONS + ".\"PART_ID\") from " + PARTITIONS + ""
+      + "  inner join " + TBLS + " on " + PARTITIONS + ".\"TBL_ID\" = " + TBLS + ".\"TBL_ID\" "
+      + "    and " + TBLS + ".\"TBL_NAME\" = ? "
+      + "  inner join " + DBS + " on " + TBLS + ".\"DB_ID\" = " + DBS + ".\"DB_ID\" "
+      + "     and " + DBS + ".\"NAME\" = ? "
+      + " where " + DBS + ".\"CTLG_NAME\" = ? and " + PARTITIONS + ".\"PART_NAME\" like ? ";
+
+    Object[] params = new Object[4];
+    params[0] = table.getTableName();
+    params[1] = table.getDbName();
+    params[2] = table.getCatName();
+    params[3] = partialName;
+
+    return getNumPartitionsViaSql(queryText, params);
+  }
+
+  private int getNumPartitionsViaSql(String queryText, Object[] params) {
```

Comment:
can we rename this method to `getCountOfQuery` and move it to `MetastoreDirectSqlUtils`?

Reply:
Addressed.